### PR TITLE
refactor yt-dlp

### PR DESCRIPTION
**Pull request by pirate486743186**
move content from youtube-dl to yt-dlp
youtude-dl and gallery-dl redirect to yt-dlp

yt-dlp has become the main program now. The future of youtube-dl is uncertain.

### CODE REVIEW

File: etc/profile-m-z/youtube-dl.profile [hunk 1]
```diff
@@ -5,63 +5,17 @@ quiet
...
+#include globals.local
 
 noblacklist ${HOME}/.cache/youtube-dl
 noblacklist ${HOME}/.config/youtube-dl
-noblacklist ${HOME}/.netrc
-noblacklist ${MUSIC}
-noblacklist ${VIDEOS}
 
 # Allow python (blacklisted by disable-interpreters.inc)
 include allow-python2.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/youtube-dl.profile:15]: included after include of disable-interpreters.inc => no-op

**Review by rusty-snake — CHANGES_REQUESTED** [review]
LGTM

The python2 include does not work.

File: etc/profile-m-z/yt-dlp.profile [hunk 2]
```diff
@@ -5,17 +5,65 @@ quiet
 # Persistent local customizations
 include yt-dlp.local
 # Persistent global definitions
-# added by included profile
-#include globals.local
+include globals.local
+
+# breaks when installed under ${HOME} via `pip install --user` (see #2833)
```
Comment (glitsj16) [inline etc/profile-m-z/yt-dlp.profile:10]: Wouldn't it make more sense to keep `noexec ${HOME}` and add a new comment instead? Users who install it like this face apparmor issues anyway if they use that.

File: etc/profile-m-z/yt-dlp.profile [hunk 3]
```diff
@@ -5,17 +5,65 @@ quiet
...
+private-tmp
 
-private-bin ffprobe,yt-dlp
-private-etc yt-dlp.conf
+dbus-user none
+dbus-system none
 
-# Redirect
-include youtube-dl.profile
+#memory-deny-write-execute - breaks on Arch (see issue #1803)
```
Comment (glitsj16) [inline etc/profile-m-z/yt-dlp.profile:68]: Is this still correct? I've been running with mdwe (on Arch Linux and Ubuntu LTS) in a local override for quite some time without apparent issues.

**Review by glitsj16 — COMMENTED** [review]
LGTM. Just a few general questions that are not related to changing the redirect logic.

File: etc/profile-m-z/yt-dlp.profile (hunk 3 above)
Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:68]: @rusty-snake has it in [kyst](https://github.com/rusty-snake/kyst/blob/main/firejail/youtube-dl.local) as well. Enabling it seems well supported, let's do it!

File: etc/profile-m-z/yt-dlp.profile (hunk 2 above)
Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:10]: I'd go for:

```
# If you installed via pip under ${HOME} add `ignore noexec ${HOME}`
# to your yt-dlp.local. AppArmor needs to allow it too, as per examples
# in /etc/apparmor.d/local/firejail-default.
noexec ${HOME}
```

File: etc/profile-m-z/youtube-dl.profile (hunk 1 above)
Comment (glitsj16) [reply etc/profile-m-z/youtube-dl.profile:15]: @pirate486743186 Looks correct to me.

File: etc/profile-m-z/yt-dlp.profile (hunk 2 above)
Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:10]: > isn't "noexec ${HOME}" already active from an inc?

I messed up here. That's correct.

> from what you said, i have no idea what to put in "/etc/apparmor.d/local/firejail-default"

https://github.com/netblue30/firejail/blob/1f5f84f9596ba7d26c2f2e35dad7728e12afaec9/etc/apparmor/firejail-local#L3-L10

Users with yt-dlp installed like that could go for the third example:
`owner @HOME/.local/bin/** ix,`

NOTE: I'll fix these examples, the comma at the end is needed in all three examples.

File: etc/profile-m-z/youtube-dl.profile (hunk 1 above)
Comment (rusty-snake) [reply etc/profile-m-z/youtube-dl.profile:15]: Your right, was comparing it in the wrong order. Shouldn't review late at night.

File: etc/profile-m-z/yt-dlp.profile (hunk 2 above)
Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:10]: This is a bit more involved than usual due to the combination of a local install and AppArmor. The thing is, if users (1) install yt-dlp via pip under /home/foo _and_ (2) are using AppArmor they can harden the Firejail sandbox by (3) adding those rules to /etc/apparmor.d/local/firejail-default. In that case they benefit from keeping the `apparmor` option.

Not sure if all that is conveyed in the current comment. To be clear, I'm not at all suggesting your comment is incorrect. I do appreciate your effort to make it as precise as possible for (Ubuntu) users. Not an easy job though to explain how to configure AA in a few lines. Maybe we can refer to distro documentation?

Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:10]: > I rather not give them a "RTFM" comment.

No problem, I can dig that. Marking conversation as resolved.

File: etc/profile-m-z/yt-dlp.profile (hunk 3 above)
Comment (glitsj16) [reply etc/profile-m-z/yt-dlp.profile:68]: > i'll add them all at once

Okay :) Marking conversation as resolved.

**Review by rusty-snake — APPROVED** [review]

**Comment by netblue30** [comment]
merged, thanks!

**Pull request merged**